client/modules/IDE/components/CollectionList/CollectionListRow.jsx

```diff
@@ -116,7 +119,10 @@ const CollectionListRowBase = (props) => {
             onKeyUp={handleRenameEnter}
             onBlur={handleRenameBlur}
             onClick={(e) => e.stopPropagation()}
-            ref={renameInput}
+            ref={(node) => {
+              renameInput.current = node;
+              handleRenameFocus();
+            }}

```

Comment:
Does the ref function get executed every time that the element is mounted?  Like if you open it and close it and open it again, will it get focused the second time?  I think so but I'm not certain off the top of my head.

Reply:
> will it get focused the second time?

Yes, it will. The way it works is when the component gets mounted the ref callback is called with the input field as an parameter. And when the component gets unmounted the callback is called again with null as an parameter.